server/routes/maintainingReports/reviewer.ts

```diff
@@ -37,10 +37,16 @@ export default class ReviewerRoutes {
       dateTo: dateTo ? parseDate(dateTo, 'DD/MM/YYYY') : null,
     })
 
+    const searchCriteria = { ...query }
+
+    if (searchCriteria.dateTo) {
+      searchCriteria.dateTo = parseDate(dateTo, 'DD/MM/YYYY').endOf('day')
```

Comment:
to avoid the additional conditional , the mutability and the duplicate parsing; could you just add a `.endOf('day')` on line 37?

Reply:
That was my first thoughts too but calling .endOf on line 37 isn't possible because parseDate() can potentially return null , i.e when date is not valid.

```
export const parseDate = (val: string | unknown, format: string): Moment | null => {
  if (!val) {
    return null
  }
  const date = moment(val, format, true)
  return date.isValid() ? date : **null**
}
```

Reply:
can you use `?.` ? 

`?.endOf('day')`

Reply:
yes. that will work.